code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm

```diff
@@ -159,6 +158,11 @@
 		finish_setup(new_character, i)
 		assigned += applicant
 		notify_ghosts("[applicant.name] has been picked for the ruleset [name]!", source = new_character, action = NOTIFY_ORBIT, header="Something Interesting!")
+	// No one got the role
+	if(!length(assigned))
+		message_admins("No players were eligible for the ruleset [name] - the previous applicants were revived/left and could no longer take the role.")
+		log_game("DYNAMIC: No players were eligible for the ruleset [name] - the previous applicants were revived/left and could no longer take the role.")
+		mode.executed_rules -= src
```

Comment:
shouldn't this cleanup and give back threat?

Reply:
It does, when execute() fails. assigned.len is check in execute(), where it returns false and triggers the cleanup.

Reply:
It appears the `executed_rules` removal is actually redundant here, since that is only added when `execute()` succeeds.